### DISCUSSION TITLE
Dukun spirit shop

### DISCUSSION THREAD

**Discussion by AmKeii23** [thatskymod/Sky-CotL-Scripts#109] [Q&A] [answered]
Bad grammars. 
Hello, im new to canvas. I have been using dukun mod for the past few weeks. 

So i will just jump in now. Dukun has the option to buy any spirit clothes/cosmetics in the Spirit Shop. So I click in the spirit shop and choose the spirit i like, but when i click buy the hair and it show "denied". 

Does buying hair require anything? I have enough candles and i have rememory(or saying that i did "collect" the memory from spirit) the spirit but still say denied. 

Can someone help?

**Comment by thatskymod** [comment, marked as answer]
Indeed! As @ObservantLurker correctly stated, only items meeting these criteria are purchasable:

a) They must be active (for limited items, this means they're officially available during a Traveling Spirit visit or event).
b) Their dependent nodes must be unlocked (for example, emotes must be purchased in the correct order: Level 1-2-3-4, including any nodes inbetween).

There are 4 possible status codes for purchases:

1. `ok`: Purchase successful
2. `denied`: Insufficient privileges to buy the item (often due to time-related restrictions, as mentioned above)
3. `insuf`: Insufficient in-game currency
4. `already`: Node already unlocked

These restrictions are enforced by the game's remote backend, which is why mods can read but not interact with these items. To provide seamless user experience, developers might consider filtering out unavailable items, though this implementation choice is up to each individual.

Hope this helps!

**Reply by ARIANNA-One** [reply]
Every single time I wanna buy from the spirit shop, it just says "I'm a teapot". The spirit shop is useless at this point. I'm not sure how to use it at all and how to buy the item I want.

**Reply by thatskymod** [reply]
Hello!

The error message you're running into, "418: I'm a teapot" is an [RFC 2324](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/418) status code, indicating that the server "won't brew coffee because it is a teapot". Thatgamecompany has adopted this status to show that their server cannot fulfill your request due to **version** or **user agent** mismatch.

Please try the following:

1. Ensure you're not attempting to buy a past or future exclusive item that isn’t currently available. Purchase an item that is currently active.
2. Deactivate **"Bypass game update"** in the Canvas settings, as this option can interfere with **version information**.

Consider opening a new thread for better exposure, as commenting on a previously resolved one may limit the visibility of your question for other readers.
